contexts/credentials/v2

```diff
@@ -5,6 +5,26 @@
 
     "id": "@id",
     "type": "@type",
+    "kid": {
```

Comment:
@OR13 kid always appears in the header.
This means that the context should be used outside of the payload where it appears?

Same comment for all JWS claims.

Reply:
`@context` is also allowed in JOSE headers, when its present, should the terms have definitions like this or not?

The core data model states a "VerifiableCredential" has "proof" and "metadata"... headers and signatures seem like proof and metadata to me... also... `kid` appears in payloads... see:

https://datatracker.ietf.org/doc/html/draft-ietf-oauth-selective-disclosure-jwt-05#appendix-A.3

https://www.rfc-editor.org/rfc/rfc7800.html#section-3.4

```
     {
      "iss": "https://server.example.com",
      "sub": "17760704",
      "aud": "https://client.example.org",
      "exp": 1440804813,
      "cnf":{
        "jku": "https://keys.example.net/pop-keys.json",
        "kid": "2015-08-28"
       }
     }
```

Omitting `kid` from this PR would cause `cnf` "RDF graph" to be build improperly for key bound SD-JWT Verifiable Credentials that are trying to leverage JSON-LD.

Reply:
Ah, I see. Fine. Let's continue under 1175 -- it's just a matter of keeping everything in 1 file or splitting it per securing mechanism.

Note that SD-JWT is defining only _sd, _sd_alg, and is not requiring cnf or any jwt claims; hence these should be defined in a separate document. (which was not the case in the previous version of sd-jwt specs)

wrt context in the jose headers -- let's discuss this in the context of JAdES that introduced custom header claims